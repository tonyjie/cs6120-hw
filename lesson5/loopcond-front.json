{
  "functions": [
    {
      "instrs": [
        {
          "label": "entry"
        },
        {
          "dest": "x",
          "op": "const",
          "type": "int",
          "value": 0
        },
        {
          "dest": "i",
          "op": "const",
          "type": "int",
          "value": 0
        },
        {
          "dest": "one",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "label": "loop"
        },
        {
          "dest": "max",
          "op": "const",
          "type": "int",
          "value": 10
        },
        {
          "args": [
            "i",
            "max"
          ],
          "dest": "cond",
          "op": "lt",
          "type": "bool"
        },
        {
          "args": [
            "cond"
          ],
          "labels": [
            "body",
            "exit"
          ],
          "op": "br"
        },
        {
          "label": "body"
        },
        {
          "dest": "mid",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "args": [
            "i",
            "mid"
          ],
          "dest": "cond",
          "op": "lt",
          "type": "bool"
        },
        {
          "args": [
            "cond"
          ],
          "labels": [
            "then",
            "endif"
          ],
          "op": "br"
        },
        {
          "label": "then"
        },
        {
          "args": [
            "x",
            "one"
          ],
          "dest": "x",
          "op": "add",
          "type": "int"
        },
        {
          "labels": [
            "endif"
          ],
          "op": "jmp"
        },
        {
          "label": "endif"
        },
        {
          "dest": "factor",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "args": [
            "x",
            "factor"
          ],
          "dest": "x",
          "op": "mul",
          "type": "int"
        },
        {
          "args": [
            "i",
            "one"
          ],
          "dest": "i",
          "op": "add",
          "type": "int"
        },
        {
          "labels": [
            "loop"
          ],
          "op": "jmp"
        },
        {
          "label": "exit"
        },
        {
          "args": [
            "x"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
