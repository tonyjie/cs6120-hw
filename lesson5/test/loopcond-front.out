Dom Frontier:
  entry: 
  loop: loop
  body: loop
  then: endif
  endif: loop
  exit: 

